File: android/app/src/main/java/com/rcforb/android/models/DataTypes.kt
package com.rcforb.android.models

import java.util.Date
import java.util.UUID

enum class ConnectionState {
    DISCONNECTED,
    AUTHENTICATING,
    AUTHENTICATED,
    CONNECTING,
    CONNECTED,
    FAILED
}

data class RemoteStation(
    val serverId: String,
    val serverName: String,
    val description: String,
    val host: String,
    val port: Int,
    val voipPort: Int,
    val online: Boolean,
    var radioInUse: Boolean,
    var radioOpen: Boolean,
    val serverVersion: String,
    val radioModel: String,
    val country: String,
    val gridSquare: String,
    val latitude: Double,
    val longitude: Double,
    val userCount: Int,
    val maxUsers: Int,
    val isV7: Boolean
)

data class RadioStateData(
    val frequencyA: Int = 0,
    val frequencyB: Int = 0,
    val buttons: Map<String, Int> = emptyMap(),
    val buttonOrder: List<String> = emptyList(),
    val dropdowns: Map<String, String> = emptyMap(),
    val dropdownLists: Map<String, List<String>> = emptyMap(),
    val dropdownOrder: List<String> = emptyList(),
    val sliders: Map<String, Double> = emptyMap(),
    val sliderRanges: Map<String, SliderRange> = emptyMap(),
    val sliderOrder: List<String> = emptyList(),
    val meters: Map<String, MeterData> = emptyMap(),
    val meterOrder: List<String> = emptyList(),
    val messages: Map<String, String> = emptyMap(),
    val messageOrder: List<String> = emptyList(),
    val statuses: Map<String, Boolean> = emptyMap(),
    val statusOrder: List<String> = emptyList(),
    val smeterA: Double = 0.0,
    val smeterALabel: String = "",
    val smeterB: Double = 0.0,
    val smeterBLabel: String = "",
    val txEnabled: Boolean = false
)

data class SliderRange(
    val min: Double,
    val max: Double,
    val step: Double,
    val displayOffset: String
)

data class MeterData(
    val value: Double,
    val max: Double,
    val unit: String
)

data class ServerInfoData(
    val serverId: String = "",
    val serverVersion: String = "",
    val serverUptime: String = "",
    val serverTime: String = "",
    val radioName: String = "",
    val radioDriver: String = "",
    val radioOpen: Boolean = false,
    val radioInUse: Boolean = false,
    val radioInUseBy: String = "",
    val tot: Int = 180
)

data class RotatorStateData(
    val bearing: Int = 0,
    val elevation: Int = 0,
    val moving: Boolean = false,
    val buttons: Map<String, Int> = emptyMap(),
    val buttonOrder: List<String> = emptyList()
)

data class AmpStateData(
    val buttons: Map<String, Int> = emptyMap(),
    val buttonOrder: List<String> = emptyList(),
    val dropdowns: Map<String, String> = emptyMap(),
    val dropdownLists: Map<String, List<String>> = emptyMap(),
    val dropdownOrder: List<String> = emptyList(),
    val sliders: Map<String, Double> = emptyMap(),
    val sliderRanges: Map<String, SliderRange> = emptyMap(),
    val sliderOrder: List<String> = emptyList(),
    val meters: Map<String, MeterData> = emptyMap(),
    val meterOrder: List<String> = emptyList()
)

data class SwitchStateData(
    val buttons: Map<String, Int> = emptyMap(),
    val buttonOrder: List<String> = emptyList()
)

data class ChatMessage(
    val id: String = UUID.randomUUID().toString(),
    val user: String,
    val text: String,
    val timestamp: Date = Date(),
    val isSystem: Boolean
)

data class AuthResult(
    val success: Boolean,
    val message: String,
    val apiKey: String? = null
)

data class SavedCredentials(
    val user: String,
    val password: String
)

File: macOS/RCForb/Sources/Models/DataTypes.swift
import Foundation

// MARK: - Connection State

enum ConnectionState: String {
    case disconnected
    case authenticating
    case authenticated
    case connecting
    case connected
    case failed
}

// MARK: - Remote Station

struct RemoteStation: Identifiable, Equatable {
    var id: String { serverId }
    let serverId: String
    let serverName: String
    let description: String
    let host: String
    let port: Int
    let voipPort: Int
    let online: Bool
    var radioInUse: Bool
    var radioOpen: Bool
    let serverVersion: String
    let radioModel: String
    let country: String
    let gridSquare: String
    let latitude: Double
    let longitude: Double
    let userCount: Int
    let maxUsers: Int
    let isV7: Bool
}

// MARK: - Radio State

struct RadioStateData {
    var frequencyA: Int = 0
    var frequencyB: Int = 0
    var buttons: [String: Int] = [:]
    var buttonOrder: [String] = []
    var dropdowns: [String: String] = [:]
    var dropdownLists: [String: [String]] = [:]
    var dropdownOrder: [String] = []
    var sliders: [String: Double] = [:]
    var sliderRanges: [String: SliderRange] = [:]
    var sliderOrder: [String] = []
    var meters: [String: MeterData] = [:]
    var meterOrder: [String] = []
    var messages: [String: String] = [:]
    var messageOrder: [String] = []
    var statuses: [String: Bool] = [:]
    var statusOrder: [String] = []
    var smeterA: Double = 0
    var smeterALabel: String = ""
    var smeterB: Double = 0
    var smeterBLabel: String = ""
    var txEnabled: Bool = false
}

struct SliderRange {
    let min: Double
    let max: Double
    let step: Double
    let displayOffset: String
}

struct MeterData {
    var value: Double
    var max: Double
    var unit: String
}

// MARK: - Server Info

struct ServerInfoData {
    var serverId: String = ""
    var serverVersion: String = ""
    var serverUptime: String = ""
    var serverTime: String = ""
    var radioName: String = ""
    var radioDriver: String = ""
    var radioOpen: Bool = false
    var radioInUse: Bool = false
    var radioInUseBy: String = ""
    var tot: Int = 180
}

// MARK: - Peripheral State

struct RotatorStateData {
    var bearing: Int = 0
    var elevation: Int = 0
    var moving: Bool = false
    var buttons: [String: Int] = [:]
    var buttonOrder: [String] = []
}

struct AmpStateData {
    var buttons: [String: Int] = [:]
    var buttonOrder: [String] = []
    var dropdowns: [String: String] = [:]
    var dropdownLists: [String: [String]] = [:]
    var dropdownOrder: [String] = []
    var sliders: [String: Double] = [:]
    var sliderRanges: [String: SliderRange] = [:]
    var sliderOrder: [String] = []
    var meters: [String: MeterData] = [:]
    var meterOrder: [String] = []
}

struct SwitchStateData {
    var buttons: [String: Int] = [:]
    var buttonOrder: [String] = []
}

// MARK: - Chat

struct ChatMessage: Identifiable {
    let id = UUID()
    let user: String
    let text: String
    let timestamp: Date
    let isSystem: Bool
}

// MARK: - Auth

struct AuthResult {
    let success: Bool
    let message: String
    var apiKey: String?
}

struct SavedCredentials {
    let user: String
    let password: String
}
